### ISSUE TITLE
RDS/Aurora cluster port does not change after terraform apply

### ISSUE THREAD

**Issue by solidspark** [terraform-providers/terraform-provider-aws#2581]
We created a Aurora Postgres cluster using a `rds_cluster` module, which had a default of port 3306. After the cluster stood up, we corrected the port by overriding the default value in the module. When we ran terraform apply, **the port does not change**. Re-running `terraform plan` shows the state did not change.

Plan:
```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ module.foo-db.aws_rds_cluster.this
      port: "3306" => "5432"


Plan: 0 to add, 1 to change, 0 to destroy.
```

Apply:
```
module.foo-db.aws_rds_cluster.this: Modifying... (ID: foo-db)
  port: "3306" => "5432"
module.foo-db.aws_rds_cluster.this: Modifications complete after 1s (ID: foo-db)

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

### Terraform Version
`Terraform v0.10.6`
`provider-aws v1.2`

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_rds_cluster

### Expected Behavior
`terraform apply` successfully changes the port or returns an error/message noting that it cannot change the port.

### Actual Behavior
`terraform apply` does not change state.

### Steps to Reproduce
1. `terraform apply`

**Comment by solidspark** [comment]
AWS has advised that the port is not modifiable on Aurora Postgres clusters. It can only be set at build time.

**Comment by apparentlymart** [comment]
Hi @solidspark! Sorry for the confusing behavior and thanks for clarifying with AWS about what is possible here.

I think the best resolution for this issue would be to mark this argument as "forces new resource", which would then cause Terraform to offer to replace the instance if the port number changes. I expect that in practice you generally wouldn't _want_ to replace the instance with Terraform, since it'd require careful steps to preserve existing data, but at least in that case the Terraform plan output would correctly reflect what needs to happen so that the operator can decide how to proceed.

**Comment by bflad** [comment]
The fix for this (#4144) has been merged into master and will release with v1.14.1 of the AWS provider, likely later today.

**Comment by bflad** [comment]
This has been released in [version 1.14.1 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1141-april-11-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
